merkletree/pad.go

```diff
@@ -76,48 +93,65 @@ func (pad *PAD) Update(policies Policies) {
 		pad.loadedEpochs = append(pad.loadedEpochs[:0], pad.loadedEpochs[n:]...)
 	}
 
-	pad.updateInternal(policies, pad.currentSTR.epoch+1)
+	pad.updateInternal(policies, pad.latestSTR.epoch+1)
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	str := pad.latestSTR
+	if str == nil {
```

Comment:
I see. This check is performed when the `PAD` is initialized, and it should be true once only when `NewPAD` is called. I presume that no later calls can lead to `pad.latestSTR == nil`.